hyperspy/drawing/marker.py

```diff
@@ -0,0 +1,288 @@
+# Copyright 2007-2011 The Hyperspy developers
+#
+# This file is part of  Hyperspy.
+#
+#  Hyperspy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  Hyperspy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  Hyperspy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+class Marker(object):
+
+    """Marker that can be added to the signal figure
+
+    Attributes
+    ----------
+    type : {'line', 'axvline', 'axhline', 'text', 'pointer', 'rect'}
+        Select the type of markers
+    orientation : {None,'v','h'}
+        Orientation for lines. 'v' is vertical, 'h' is horizontal.
+    marker_properties : dictionary
+        Accepts a dictionary of valid (i.e. recognized by mpl.plot)
+        containing valid line properties. In addition it understands
+        the keyword `type` that can take the following values:
+        {'line', 'text'}
+
+    Methods
+    -------
+    set_marker_properties
+        Enables setting the line_properties attribute using keyword
+        arguments.
+    set_data
+        Set the data in a structured array. Each field of data should have
+        the same dimensions than the nagivation axes. Some fields need to be
+        defined depending on the type.
+        For 'line': 'x1','y1','x2','y2' (All of them if orientation
+            is None)
+        For 'axvline': 'x1'
+        For 'axhline': 'y1'
+        For 'text': 'x1','y1','text'
+        For 'pointer': 'x1','y1','size'. 'size' is optional
+        For 'rect': 'x1','y1','x2','y2'
+
+    Example
+    -------
+
+    >>> s = signals.Spectrum(random.random([10, 100]))
+    >>> m = utils.plot.marker()
+    >>> m.type = 'axvline'
+    >>> m.set_marker_properties(color='green')
+    >>> m.set_data(x1=range(10))
+    >>> s.plot()
+    >>> s._plot.signal_plot.add_marker(m)
+    >>> m.plot()
+
+    >>> im = signals.Image(random.random([10, 50, 50]))
+    >>> m = utils.plot.marker()
+    >>> m.type = 'text'
+    >>> m.set_marker_properties(fontsize = 30, color='red')
+    >>> m.set_data(x1=range(10), y1=range(10)[::-1], text='hello')
+    >>> im.plot()
+    >>> im._plot.signal_plot.add_marker(m)
+    >>> m.plot()
+
+    >>> im = signals.Image(np.zeros((100, 100)))
+    >>> m = utils.plot.marker()
+    >>> m.type = 'rect'
+    >>> m.set_marker_properties(linewidth=4, color='red', ls='dotted')
+    >>> m.set_data(x1=20, x2=70, y1=20, y2=70)
+    >>> im.plot()
+    >>> im._plot.signal_plot.add_marker(m)
+    >>> m.plot()
+
+    """
+
+    def __init__(self):
```

Comment:
How about adding ```type``` field for init?
I noticed throughout the code that you always have to first create the marker, and then change type to what you wish. In this case it can be done in a single line, and at the same time have an obvious (from tab-completion) default value